lib/ex_unit/lib/ex_unit/assertions.ex

```diff
@@ -614,7 +614,6 @@ defmodule ExUnit.Assertions do
   def __pins__(pins) do
     pins
     |> Enum.filter(fn {{_, ctx}, _} -> ctx == nil end)
```

Comment:
If it is a map, we probably need to explicitly sort it, order is not guaranteed:

```suggestion
    |> Enum.filter(fn {{_, ctx}, _} -> ctx == nil end)
    |> Enum.sort()
```